File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/RestartPacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RestartPacket implements ServerPacket {
  private int retryMs;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_RESTART;
  }

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    writer.writeInt32(stream, retryMs);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/RestartPacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.core.BanchoPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataWriter

class RestartPacket(var retryMs: Int = 0) : BanchoPacket(), BanchoPacket.Server {
    override val type = PacketType.BANCHO_RESTART

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeInt32(out, retryMs)
    }
}
